src/app/shared/components/workshop-card/workshop-card.component.html

```diff
@@ -1,22 +1,52 @@
+<div *ngIf="((isMainPage$ | async) || user.role !== role.provider) then UserActions; else ProviderActions">
```

Comment:
it could be optimized by moving isMainPage subscription outside workshop card component and passing it as input parameter

Reply:
@dmitryy90 genialno))))))